pandas/tests/io/test_excel.py

```diff
@@ -119,6 +119,15 @@ def get_exceldf(self, basename, ext, *args, **kwds):
 class ReadingTestsBase(SharedItems):
     # This is based on ExcelWriterBase
 
+    @pytest.fixture(autouse=True, params=['xlrd', None])
+    def set_engine(self, request):
+        func_name = "get_exceldf"
+        old_func = getattr(self, func_name)
+        new_func = partial(old_func, engine=request.param)
```

Comment:
Well, there isn't much to refactor for that method (it's only two lines), but a greater reorganization would indeed be nice since we're introducing more `pytest`-like code into this file (perhaps take inspiration from the work I did with the `read_csv` tests).

Reply:
Yea by refactor I meant more-so replace with a fixture or something besides an instance method on a base class